### PR TITLE
Marking format_on_save timeout configurable.

### PR DESCRIPTION
**Pull request by thisduck**
Since `format_on_save` can not be overwritten now, I've added a configuration param for the timeout.

### CODE REVIEW

**Comment by junnplus** [comment]
You can still use `on_attach` to overwrite the default.

File: lua/nvim-lsp-setup/utils.lua [hunk 1]
```diff
@@ -40,13 +40,13 @@ function M.disable_formatting(client)
     end
 end
 
-function M.format_on_save(client)
+function M.format_on_save(client, opts)
```
Comment (junnplus) [inline lua/nvim-lsp-setup/utils.lua:43]: Thanks, we can add a `timeout` argument instead of a headcode.

Comment (junnplus) [inline lua/nvim-lsp-setup/utils.lua:43]:
```suggestion
function M.format_on_save(client, timeout)
```

File: README.md [hunk 2]
```diff
@@ -68,6 +68,8 @@ require('nvim-lsp-setup').setup({
     -- gi = 'lua require"telescope.builtin".lsp_implementations()',
     -- gr = 'lua require"telescope.builtin".lsp_references()',
     mappings = {},
+    -- configurable format_on_save timeout
+    format_on_save_timeout = 1000,
```
Comment (junnplus) [inline README.md:72]: IMHO, we can still use `on_attach` instead of a new option.

WDYT?

Comment (thisduck) [reply README.md:72]: Yes, that would be preferable, but it seems like `format_on_save` is called outside the `on_attach` now. https://github.com/junnplus/nvim-lsp-setup/blob/main/lua/nvim-lsp-setup/init.lua#L11

Seems like this commit changed it: https://github.com/junnplus/nvim-lsp-setup/commit/ceefba1938601e9609c39aba12b269f99b6a5dcd

Can you revert that so it's called inside? Thanks!

Comment (thisduck) [reply README.md:72]: I added a PR for that: https://github.com/junnplus/nvim-lsp-setup/pull/26

Comment (junnplus) [reply README.md:72]: I don't understand why to revert it, the default is to support formatting on save.

If you want to change the timeout of format, just set `on_attach` to overwrite the default.

Comment (junnplus) [reply README.md:72]: Thanks

**Pull request closed without merging**